src/lib/handleRef.ts

```diff
@@ -20,7 +20,7 @@ const handleRef = <N>(ref: React.Ref<N>, node: N) => {
     return
   }
 
-  if (typeof ref === 'object') {
+  if (ref !== null && typeof ref === 'object') {
```

Comment:
could you, please, suggest when this condition is critical for us to check? What is the case that could lead to that? Is it some sort of 'race condition' during component's unmount that we would like to prevent with that?

Thing that I am worrying about is that it could lead to memory leaks. Suppose that there is a parent component that will capture a dom element of its child component:

```jsx
<Parent>
  <Child innerRef={domNode => this.childDomNode = domNode} />
</Parent>
```

In that case `Parent` might never update `this.childDomNode` to `null`, even if `Child` will be unmounted (without this check, previously, `this.childDomNode` will be updated to `null`).